dsf-bpe/dsf-bpe-process-api-v2/src/main/java/dev/dsf/bpe/v2/service/MimetypeService.java

```diff
@@ -3,23 +3,89 @@
 import java.io.ByteArrayInputStream;
 import java.io.InputStream;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public interface MimetypeService
 {
+	Logger logger = LoggerFactory.getLogger(MimetypeService.class);
+
+	record ValidationResult(String declaredBaseType, String declaredSubType, String detectedBaseType,
+			String detectedSubType)
+	{
+		public String declared()
+		{
+			return declaredBaseType + "/" + declaredSubType;
+		}
+
+		public String detected()
+		{
+			return detectedBaseType + "/" + detectedSubType;
+		}
+
+		public boolean mimetypesMatch()
+		{
+			return declared().equals(detected());
+		}
+	}
+
 	/**
 	 * Detects the mimetype of the provided byte array and validates if the detected mimetype equals the declared
-	 * mimetype. Logs a warning if the full mimetypes do not match, throws a {@link RuntimeException} if the base
-	 * mimetypes do not match.
+	 * mimetype. Returns a {@link ValidationResult} containing both the declared and detected MIME types. This result
+	 * can be used to drive custom logic based on whether the detected type matches the declared type.
+	 *
+	 * @param stream
+	 *            input stream of which the mimetype should be detected
+	 * @param declared
+	 *            the declared mimetype of the data
```

Comment:
Since we are using strings and no complex types for the MIME type parameters, we may want to add an example.
```suggestion
	 * @param declared
	 *            the declared MIME type of the data, e.g. <code>"application/pdf"</code>
```
Generally I think we should write "MIME type" instead of "mimetype" in javadoc as MIME is originally an acronym for "Multipurpose Internet Mail Extensions".